mono/metadata/timer-ms.c

```diff
@@ -0,0 +1,332 @@
+
+#include <glib.h>
+
+#include "timer-ms.h"
+
+#include "class-internals.h"
+#include "gc-internal.h"
+#include "object.h"
+#include "object-internals.h"
+#include "threads-types.h"
+#include "threadpool-internals.h"
+#include "utils/atomic.h"
+#include "utils/mono-time.h"
+#include "utils/mono-semaphore.h"
+
+enum {
+	STATUS_NOT_INITIALIZED,
+	STATUS_INITIALIZING,
+	STATUS_INITIALIZED,
+	STATUS_CLEANING_UP,
+	STATUS_CLEANED_UP,
+};
+
+enum {
+	TIMER_STATUS_REQUESTED,
+	TIMER_STATUS_WAITING_FOR_REQUEST,
+	TIMER_STATUS_NOT_RUNNING,
+};
+
+typedef struct {
+	MonoDomain *domain;
+	guint32 next_due_ticks;
+	guint32 last_fire_ticks;
+} TimerDomain;
+
+typedef struct {
+	MonoInternalThread *thread;
```

Comment:
This field is assigned once and never read.
This is pointing to managed memory and it's not been registered as root.
We're better off removing it.